doc/design.md

```diff
@@ -273,6 +295,121 @@ Returns:
     * tbd
 
 
+### **`POST`** `/oeci_login` ?
+
+## Question
+
+What's the workflow for accessing OECI?
+
+Is it a webpage redirect, like the frontend prototype?
+
+Or is it an authentication serverside, like the current Crawler implementation?
+These behaviors need to be reconciled.
+
+I see three design versions:
+
+- Rely on a frontend page redirect, which obtains login access from the OECI directly to allow subsequent requests by Crawler. This requires a Crawler redesign, which currently passes creds to the remote site. It also requires caching a login token provided by OECI. Does the OECI login allow that, e.g. use a cookie?
+
+- Roll the login  into the `/search/` endpoint.
+If so, we need to add oeci credentials to the **`POST`** `/search/ ` body. And also cache the login credentials clientside so we can re-send them on every search (bad).
+
+- Send oeci login credentials to the backend once, and the /search/ endpoint uses them for every /search/ call. That means caching the OECI login creds in the backend.
+
+
+
+### **`POST`** `/api/forgot_password`
+
+Requests a password reset link be sent to a valid (recognized) email address
+
+`POST` body:
+
+- format: `JSON`
+- fields:
+    * email
+
+Status codes:
+
+- `200 OK`
+- `422 UPROCESSABLE ENTITY` email not recognized
+- `5xx` of some kind if the email-send failed
+
+
+### **`POST`** `/api/update_password`
+
+Follows a password reset link (from a sent email), that provides login authorization
+
+## Question: How does this work?
+
+A password reset link provided in an email needs to be a GET request, so the endpoint requires a different method for yielding auth access. If this method is a POST and auth-protected, we need to first obtain an auth_token via the **`GET`** `/api/auth_token` method below.
```

Comment:
Not sure I totally understand the approach suggested here. This would be good to discuss at the next meeting.

Because we are using a JWT for authentication, and the JWT needs to be stored in our React app in a certain way for the frontend to think it's authenticated, Javascript needs to be involved before our app can become authenticated in the usual way.

Reply:
Ah, yep I see what I was leaving out here. I agree, let's go over this in a meeting.